front/lib/api/assistant/conversation.ts

```diff
@@ -677,16 +701,29 @@ export async function* postUserMessage(
   }
 
   const results = await Promise.all([
-    removeNulls(
-      await Promise.all(
-        mentions.filter(isAgentMention).map((mention) => {
-          return getLightAgentConfiguration(auth, mention.configurationId);
-        })
-      )
+    await Promise.all(
+      mentions.filter(isAgentMention).map((mention) => {
+        return getLightAgentConfiguration(auth, mention.configurationId);
+      })
     ),
     await createOrUpdateParticipation(),
   ]);
-  const agentConfigurations = results[0];
+
+  // casting non-null is ok here because we check right below that there's no
+  // nulls and err if there is
+  const agentConfigurations = results[0] as LightAgentConfigurationType[];
```

Comment:
Not sure to understand why we need this 🤔. The code casts `results[0]` to `LightAgentConfigurationType[]` assuming it's non-null, but then checks for nulls afterward 🤯. Instead of casting immediately, consider handling potential null values first. This would improve clarity and prevent assumptions.

Reply:
Right, thought the comment would clarify but true it's better below :+1: